### ISSUE TITLE
Duplicate peptide elements with similar modifications

### ISSUE THREAD

**Issue by glormph** [MSGFPlus/msgfplus#91]
Hi,

I'm on 2019.07.03 and running into trouble with duplicate XML entries (like #49, which crash percolator) in a search with both Acetyl and Trimethyl (and some other) mods. My mod file looks as follows:
```
NumMods=4
229.162932,*,fix,N-term,TMT6plex
229.162932,K,fix,any,TMT6plex
-187.152367,K,opt,any,Acetyl
-215.147282,K,opt,any,Methyl
-201.131632,K,opt,any,Dimethyl
-187.115982,K,opt,any,Trimethyl
C2H3N1O1,C,fix,any,Carbamidomethyl
O1,M,opt,any,Oxidation
```
Note that I am "cheating" since the TMT is fixed and therefore the variable mods on K are specified as the  `-(tmtmass - actual_modmass)`. 

The output is that I have two `<Peptide id="Pep_LDYNNIPTVVFSHPPIGTVGLTEDEAIHK-187">`. In one of them the -187 represents Acetyl, and in the other one it represents Trimethyl. I wonder if this negative modification faking is a good way to go about it, but the alternative of specifying actual mods would also be two mods with the same mass integer before the decimal point (and I wouldnt be able to use TMT fixed?).

I'll put the Fasta and mzML online by tomorrow if that is necessary.

```msgf_plus -Xmx8G -d db.fa -s spectra.mzML -o "out.mzid" -thread 8 -mod acetylmods.txt -tda 0 -maxMissedCleavages -1 -t 10.0ppm  -ti -1,2 -m 0 -inst 3 -e 1 -protocol 4 -ntt 2 -minLength 7 -maxLength 50 -minCharge 2 -maxCharge 6 -n 1 -addFeatures 1```

**Comment by alchemistmatt** [comment]
I would be hesitant to use two mod masses that are so close together: -187.152367 and -187.115982 are just 0.036 Da apart, or 18 ppm at 2000 m/z.  Still, you raise a valid point: by basing the peptide IDs on the integer value of the mod masses, we're open to name collisions.  We'll look into what it would take to check for this type of collision when creating the .mzid file and account for it.

**Comment by glormph** [comment]
Thanks! In the meantime I'll just not combine the two modifications :).

**Comment by FarmGeek4Life** [comment]
Fixed in release v2020.01.15 (binaries are not yet uploaded)
